lua/entities/gmod_wire_expression2/core/extpp.lua

```diff
@@ -1,225 +1,146 @@
 AddCSLuaFile()
 
--- some constants --
-
-local p_typename = "[a-z][a-z0-9]*"
-local p_typeid = "[a-z][a-z0-9]?[a-z0-9]?[a-z0-9]?[a-z0-9]?"
-local p_argname = "[a-zA-Z][a-zA-Z0-9_]*"
-local p_funcname = "[a-z][a-zA-Z0-9]*"
-local p_func_operator = "[-a-zA-Z0-9+*/%%^=!><&|$_%[%]]*"
-
-local OPTYPE_FUNCTION
-local OPTYPE_NORMAL = 0
-local OPTYPE_DONT_FETCH_FIRST = 1
-local OPTYPE_ASSIGN = 2
-local OPTYPE_APPEND_RET = 3
-
-local optable = {
-	["operator+"] = "add",
-	["operator++"] = { "inc", OPTYPE_DONT_FETCH_FIRST },
-	["operator-"] = "sub",
-	["operator--"] = { "dec", OPTYPE_DONT_FETCH_FIRST },
-	["operator*"] = "mul",
-	["operator/"] = "div",
-	["operator%"] = "mod",
-	["operator^"] = "exp",
-	["operator="] = { "ass", OPTYPE_ASSIGN },
-	["operator=="] = "eq",
-	["operator!"] = "not",
-	["operator!="] = "neq",
-	["operator>"] = "gth",
-	["operator>="] = "geq",
-	["operator<"] = "lth",
-	["operator<="] = "leq",
-	["operator&"] = "and",
-	["operator&&"] = "and",
-	["operator|"] = "or",
-	["operator||"] = "or",
-	["operator[]"] = "idx", -- typeless op[]
-	["operator[T]"] = { "idx", OPTYPE_APPEND_RET }, -- typed op[]
-
-	["operator_is"] = "is",
-	["operator_neg"] = "neg",
-	["operator_band"] = "band",
-	["operator_bor"] = "bor",
-	["operator_bxor"] = "bxor",
-	["operator_bshl"] = "bshl",
-	["operator_bshr"] = "bshr",
+local p_typename = "%l[%l%d]*"
+local p_typeid = "%l[%l%d]?[%l%d]?"
+local p_argname = "%a[%w_]*"
+local p_funcname = "%l[%w_]*"
+local p_func_operator = "[%-%+%*%%%/%^%=%!%>%<%&%|%$%[%]%w_]*"
+
+local Operators = {
+	["operator+"] = "add", ["operator-"] = "sub",
+	["operator*"] = "mul", ["operator/"] = "div",
+	["operator%"] = "mod", ["operator^"] = "exp",
+	["operator=="] = "eq", ["operator>"] = "gth",
+	["operator>="] = "geq", ["operator<"] = "lth",
+	["operator<="] = "leq", ["operator_neg"] = "neg",
+	["operator_band"] = "band", ["operator_bor"] = "bor",
+	["operator_bxor"] = "bxor", ["operator_bshl"] = "bshl",
+	["operator_bshr"] = "bshr", ["operator_is"] = "is"
 }
 
--- This is an array for types that were parsed from all E2 extensions.
-local preparsed_types
+local RemovedOperators = {
+	["operator!"] = true, ["operator!="] = true, -- These now use operator_is and operator== internally
+	["operator&"] = true, ["operator&&"] = true, -- Despite && being binary AND in E2, the preprocessor used to handle this as logical AND.
+	["operator|"] = true, ["operator||"] = true, -- Despite || being binary OR in E2, the preprocessor used to handle this as logical OR.
```

Comment:
@Vurv78 why are both operator& and operator&& listed here?

Reply:
[Because both were valid to be used as logical and / or by the extension preprocessor](https://github.com/wiremod/wire/blob/aa80d45e47ed5bb08abf2e6c38a06d8967e44067/lua/entities/gmod_wire_expression2/core/extpp.lua#L34-L37)

Reply:
and now you removed both of them?
Is there something I don't understand?

Reply:
oh I get it, you removed the ability to overload operator&/| entirely and you're just doing and/or at the Lua level, right? Probably a good idea, resolving.

Reply:
Yeah `and` / `or` now use the existing `is` operator internally to avoid needing a ton of overloads.

So all of those in the `RemovedOperator` table make use of other operators internally and aren't used themselves, so you only really need to define `is` and `eq` if you want to make a type now (would automatically make `&`, `|`, `!`, `!=` work)

This might be slower but I don't think by much, and I think it'd just be a better developer experience to not have to define 4+ more overloads than necessary